test/provisioner/provision_azure.go

```diff
@@ -326,16 +269,51 @@ func (p *AzureCloudProvisioner) CreateCluster(ctx context.Context, cfg *envconf.
 		return err
 	}
 
-	cluster, err := pollerResp.Result(ctx)
+	clusterpre, err := pollerResp.Result(ctx)
 	if err != nil {
 		return fmt.Errorf("getting cluster object: %w", err)
 	}
 
-	aksOIDCIssuer := *cluster.Properties.OidcIssuerProfile.IssuerURL
+	aksOIDCIssuer := *clusterpre.Properties.OidcIssuerProfile.IssuerURL
 	if err := createFederatedIdentityCredential(aksOIDCIssuer); err != nil {
 		return fmt.Errorf("creating federated identity credential: %w", err)
 	}
 
+	cluster, err := AzureProps.ManagedAksClient.Get(context.Background(), AzureProps.ResourceGroupName, AzureProps.ClusterName, nil)
+	if err != nil {
+		return fmt.Errorf("Failed to get AKS cluster: %q details: %w", AzureProps.ClusterName, err)
+	}
+
+	// Fetch aks-rg details
+	aks_rg := *cluster.Properties.NodeResourceGroup
```

Comment:
Can't we get this from the existing AKS object? Or we can always generate this name, thus reducing the number of calls that happen to Azure API. 

https://learn.microsoft.com/en-us/azure/aks/faq#why-are-two-resource-groups-created-with-aks